gc/base/MemoryPoolLargeObjects.hpp

```diff
@@ -115,9 +116,40 @@ class MM_MemoryPoolLargeObjects : public MM_MemoryPool {
 	double resetTargetLOARatio(MM_EnvironmentBase*);
 	void resetLOASize(MM_EnvironmentBase*, double newLOARatio);
 	void redistributeFreeMemory(MM_EnvironmentBase*, uintptr_t newOldAreaSize);
-	bool isSizeEnoughForLOA(MM_EnvironmentBase* env, uintptr_t newOldAreaSize)
+
+	/* check if new LOA size is enough for largeObjectMinimumSize and set _loaSize, _soaSize, _currentLOARatio and _currentLOABase */
+	bool isSizeEnoughForLOA(MM_EnvironmentBase* env, uintptr_t oldAreaSize, uintptr_t newLOASize, double newLOARatio, void* newLOABase=NULL)
 	{
-		return (newOldAreaSize >= _extensions->largeObjectMinimumSize);
+		bool ret = newLOASize >= _extensions->largeObjectMinimumSize;
+		if (!ret) {
+			/* No.. make LOA empty as not even big enough for one free chunk */
+			_loaSize = 0;
+			_soaSize = oldAreaSize;
+			_currentLOARatio = 0;
+			_currentLOABase = LOA_EMPTY;
+		} else {
+			_loaSize = newLOASize;
+			/* ..and SOA is whats left after LOA allocation */
+			_soaSize = oldAreaSize - _loaSize;
+			if (0 != newLOARatio) {
+				_currentLOARatio = newLOARatio;
+			} else {
+				_currentLOARatio = ((double)_loaSize) / (_loaSize + _soaSize);
```

Comment:
I'd use here 'divide by oldAreaSize' (which as mentioned before would be fetched into a method local directly from SubSpace rather than passed as an argument.
Since we  modified _loaSize and _soaSize, we can assert the sum is equal to oldAreaSize, after the mutation and before entering this if statement.